### PR TITLE
Fixes CAAM memory allocation

### PR DESCRIPTION
**Pull request by clementfaure**
Hello,

As discussed with @jenswi-linaro, here are the first two commits of PR https://github.com/OP-TEE/optee_os/pull/4404

Thanks!

### CODE REVIEW

File: core/drivers/crypto/caam/utils/utils_mem.c [hunk 1]
```diff
@@ -51,21 +51,27 @@
 static void *mem_alloc(size_t size, uint8_t type)
 {
 	void *ptr = NULL;
+	size_t cacheline_size = dcache_get_line_size();
```
Comment (jenswi-linaro) [inline core/drivers/crypto/caam/utils/utils_mem.c:54]: The return value of `dcache_get_line_size()` is only `if (type & MEM_TYPE_ALIGN)`, please move the function call to that block.

**Review by jenswi-linaro — COMMENTED** [review]
With my comment addressed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by clementfaure** [comment]
@jenswi-linaro 
Moved ```size_t cacheline_size = dcache_get_line_size();``` to ```if (type & MEM_TYPE_ALIGN)``` block
Tags applied

File: core/drivers/crypto/caam/utils/utils_mem.c (hunk 1 above)
Comment (clementfaure) [reply core/drivers/crypto/caam/utils/utils_mem.c]: Fixed in https://github.com/OP-TEE/optee_os/pull/4407

Comment (clementfaure) [reply core/drivers/crypto/caam/utils/utils_mem.c]: Fixed in https://github.com/OP-TEE/optee_os/pull/4407

**Pull request merged**